contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+
+In the table below we propose how the resource resizing directives for vertical scaling can be applied in conjunction with the 'StatefulSet' update strategy.
+
+|resizePolicy (resizeAction)|OnDelete|RollingUpdate|
+|---|---|---|
+|RestartOnly (or Action=Restart)|Resize resource with restart (with delete command)|Resize resource with restart (with allowed spec update commands)|
+|LiveResizeable (and Action=LiveResize)|Resize resource live only (with allowed spec update commands)|Resize resource live only (with allowed spec update commands)|
+LiveResizeable (and Action=LiveResizePreferred or no Action specified)|Resize resource with live resize preferred (with allowed spec update commands), i.e., resize resource with restart (with delete command) if not able to resize live.|Resize resource with live resize preferred (with allowed spec update commands), i.e., resize resource with restart (managed by controller) if not able to resize live.|
+
+## Desired Approach
+
+Any valid method to update the pod spec should be applicable for vertical scaling, e.g., using kubectl commands set, patch, apply, edit.
+Logs associated with the pod will capture the failure/success of the resize command.
+The controller will continue to attempt the update to the spec while there is a difference between the current size and size in updated spec.
+If an update is partially successful, user can know this from the logs and attempt to rectify the situation by submitting new updates (that can restore original size(s) or go for a size feasible on all the nodes). 
+
+The policy for vertical scaling can itself be mutable.
+It is desirable to submit a change in policy, confirm its acceptance (by re-reading the changed pod spec) and then submit a change to a resource size impacted by that policy. 
+
+# Design and Implementation
+
+This section highlights some of implementation details by describing API changes and the workflow of vertical scaling on a pod.
+
+## Changes on API and key components
+
+This section describes briefly API changes for pod-level vertical scaling and the related changes on the 'API server', the 'Scheduler', and the 'Kubelet'.
+
+* **ResizeRequest**
+
+A new data structure, ResizeRequest, is added to v1.PodSpec:
+
+```go
+const (
+    ResizeRequested ResizeStatus = "Requested"
+    ResizeAccepted ResizeStatus = “Accepted”
+    ResizeRejected ResizeStatus = “Rejected”
+    ResizeNone      ResizeStatus = "None"
+)
```

Comment:
suggest add some comments here, have no idea what `ResizeNone` mean

Reply:
This embedded "transaction" is a very different concept from most everything else, isn't it?  Do we have precedent for this?  It needs to be throroughly considered for API..

Reply:
To be honest, I could not find any precedent, this is just what I worry about @deads2k could you please give thoughts here?

Reply:
I feel more like the semantic is "the resource request for this pod WILL change, or the pod will die".  The transaction is not needed